offset_manager.go

```diff
@@ -250,8 +251,12 @@ func (pom *partitionOffsetManager) fetchInitialOffset(retries int) error {
 
 	switch block.Err {
 	case ErrNoError:
-		pom.offset = block.Offset
-		pom.metadata = block.Metadata
+		if block.Offset >= 0 {
+			pom.offset = block.Offset
+			pom.metadata = block.Metadata
+		} else {
+			pom.offset = pom.parent.conf.Consumer.Offsets.Initial
```

Comment:
you don't actually want to return `OffsetNewest` here, you want to ask the broker what the earliest available offset is and return that.